ios/src/TangramMap/TGGeoPolygon.h

```diff
@@ -0,0 +1,88 @@
+//
+//  TGGeoPolygon.h
+//  tangram
+//
+//  Created by Karim Naaji on 10/27/16.
+//
+//
+
+#import <Foundation/Foundation.h>
+#import "TGGeoPoint.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ Helper class to contain a polygon geometry for use in `-[TGMapViewController markerSetPolygon:polygon:]`.
+
+ The polygon winding order and internal polygons must be set according to the <a href="http://geojson.org/geojson-spec.html#polygon">
+ GeoJSON specification</a>.
+ */
+@interface TGGeoPolygon : NSObject
+
+/**
+ Inits a polygon and allocates enough memory to hold `size` geographic coordinates in the polygon paths.
+ */
+- (instancetype)initWithSize:(unsigned int)size;
+
+/**
+ Starts a new polygon path for a given `TGGeoPoint`.
+
+ @param latlon the coordinate to start a polygon path
+ */
+- (void)startPath:(TGGeoPoint)latlon;
+
+/**
+ Starts a new polygon path for a given `TGGeoPoint` and allocate enough memory to hold `size` coordinates
+ in the polygon geometry.
+
+ @param latlon the coordinate to start a polygon path
+ @param size the size of the polygon path
+ */
+- (void)startPath:(TGGeoPoint)latlon withSize:(unsigned int)size;
+
+/**
+ Adds a single coordinate to the current polygon path.
+
+ @param latlon the coordinate to add to the current path
+
+ @note This operation is a no-op if no polygon path were already started.
+ */
+- (void)addPoint:(TGGeoPoint)latlon;
+
+/**
+ Returns a pointer to a sequence of `TGGeoPoint` with a total array length equal to `-[TGGeoPolygon count:]`.
+
+ @return a pointer to the list of polygons coordinates
+ */
+- (TGGeoPoint*)coordinates;
+
+/**
+ The list of polygon `rings` contained in this geometry.
+ The sum of the values in `-[TGGeoPolygon rings:]` sum up to `-[TGeoPolygon count:]`.
+
+ @return a pointer to the list of polygon rings.
+ */
+- (int*)rings;
```

Comment:
The user shouldn't have to use this, it should be fine to leave this one as it is since I use it only to directly interface with the C++ API of Tangram. There could be another helper accessor using `NSArray` for convenience if we find the need to.

This method is only used internally, it is private/internal for the framework but still exposed publicly for simplicity. I'm planing to add a `TGGeoPolygon+Internal.h` as described [here](http://stackoverflow.com/questions/19095716/hiding-properties-from-public-framework-headers-but-leaving-available-internall) that has those methods used only in the scope of the framework. Let me know your thoughts on this!